internal/databases/mysql/mysql.go

```diff
@@ -80,6 +90,34 @@ func getLastUploadedBinlog(folder storage.Folder) (string, error) {
 	return name, nil
 }
 
+func getLastUploadedBinlogBeforeGTID(folder storage.Folder, gtid string, flavor string) (string, error) {
+	gtidParsed, err := gomysql.ParseMysqlGTIDSet(gtid)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse server gtid: %s: %v", gtid, err)
+	}
+	folder = folder.GetSubFolder(BinlogPath)
+	logFiles, _, err := folder.ListFolder()
+	if err != nil {
+		return "", err
+	}
+	sort.Slice(logFiles, func(i, j int) bool {
+		return logFiles[i].GetLastModified().Before(logFiles[j].GetLastModified())
+	})
+	if len(logFiles) == 0 {
+		return "", nil
+	}
+	for i := len(logFiles) - 1; i > 0; i-- {
+		prevGtid, err := GetBinlogPreviousGTIDsRemote(folder, logFiles[i].GetName(), flavor)
+		if err != nil {
+			return "", err
+		}
+		if gtidParsed.Contain(prevGtid) {
+			return utility.TrimFileExtension(logFiles[i].GetName()), nil
+		}
+	}
+	return "", nil
```

Comment:
Fallback to first binlog?

Reply:
> Fallback to first binlog?

First one it not better than the last one. 
In the sense, that there are still some transactions missed in base backup.
But we will need to apply all uploaded binlogs during PITR and will still fail slowly. 
Let's fail fast and keep empty `BinLogStart` to simplify debugging.

BTW, added warning for this case.